doc/source/howtos/crash-recovery.rst

```diff
@@ -0,0 +1,295 @@
+   140814 0:42:13 [Note] WSREP: Victim thread:
+   THD: 7, mode: local, state: executing, conflict: cert failure, seqno: -1
+   SQL: insert into t values (1)
+
+As soon as the other nodes become available, the cluster is formed again
+automatically. If node B and node C were just network-severed from
+node A, but they can still reach each other, they will keep functioning as
+they still form the quorum.
+
+If node A and node B crashed, you need to do enable the primary component on
+node C manually, before you can bring up node A and node B. The command to do
+this is:
+
+.. code-block:: guess
+
+   SET GLOBAL wsrep_provider_options='pc.bootstrap=true';
+
+This approach only works if the other nodes are down before doing that!
+Otherwise, you end up with two clusters having different data.
+
+
+.. admonition:: Cross Reference
+
+   :ref:`add-node`
+
+.. _pxc-crash-recovery-scenario-6:
+
+Scenario: All nodes went down without a proper shutdown procedure
+================================================================================
+
+This scenario is possible in case of datacenter power failure or when hitting a
+MySQL or Galera bug. Also it may happen as a result of data consistency being
+compromised where the cluster detects that each node has different data. The
+:file:grastate.dat` file is not updated and does not contain a valid sequence
+number (seqno). It may look like this:
+
+.. code-block: text
+
+   cat /var/lib/mysql/grastate.dat
+   # GALERA saved state
+   version: 2.1
+   uuid: 220dcdcb-1629-11e4-add3-aec059ad3734
+   seqno: -1
+   cert_index:
+
+In this case, you cannot be sure that all nodes are consistent with each
+other. Before starting :command:`mysql` daemon on any node, you have to extract
+the last sequence number by checking its transactional state.
+
+In recent Galera versions, the option :variable:`pc.recovery` (enabled by
+default) saves the cluster state into a file named :file:`gvwstate.dat` on each
+member node. As the name of this option suggests (pc – primary component), it
+saves only a cluster being in the PRIMARY state. An example content of : file
+may look like this:
+
+.. code-block:: bash
+
+   cat /var/lib/mysql/gvwstate.dat
+   my_uuid: 76de8ad9-2aac-11e4-8089-d27fd06893b9
+   #vwbeg
+   view_id: 3 6c821ecc-2aac-11e4-85a5-56fe513c651f 3
+   bootstrap: 0
+   member: 6c821ecc-2aac-11e4-85a5-56fe513c651f 0
+   member: 6d80ec1b-2aac-11e4-8d1e-b2b2f6caf018 0
+   member: 76de8ad9-2aac-11e4-8089-d27fd06893b9 0
+   #vwend
+
+We can see a three node cluster with all members being up. Thanks to this new
+feature, the nodes will try to restore primary the component once all the
+members start to see each other. This makes the PXC cluster automatically
+recover from being powered down without any manual intervention! In the logs we
+will see:
+
+.. code-block:: text
+
+   140823 15:28:55 [Note] WSREP: restore pc from disk successfully
```

Comment:
just realized that these are old logs. The formatting has changed for 8.0.  Should get newer logs (+ some of the logging messages may have changed).

Reply:
Unless the output of logs is significant, i would suggest removing the log output in this case altogether.

Reply:
ok